### DISCUSSION TITLE
Dynamic routing with dots and dashes

### DISCUSSION THREAD

**Discussion by maratsh** [vercel/next.js#15182] [Help]
We have many pages like this: "/some-page-variable-length-123123.html"
the number part is used later to do external api fetches
To match this page i've created file `pages/[id].html.js"

This path always throws 404 in dev mode (npm run start). I've also tried other options: `pages/[id]-html.js`, `pages/[id].test.js`, `pages/test-[id].js`. They all won't work.

**Comment by tarizer** [comment]
You can use [rewrites ](https://nextjs.org/docs/api-reference/next.config.js/rewrites)

```javascript
  async rewrites() {
    const rewriteArray = [
      {       
        source: "/:id.html",
        destination: "/any/path/you/want/:id",
      },
    ];

    return rewriteArray;
  },
```

**Reply by pro2s** [reply]
@tarizer thanks for advice, so
this will work if we go to the URL directly, but if we have getServerSideProps on the page and try to use `router.push('route.html')`, then instead of receiving props from the server, we will get a redirect with a page reload and a repeated request for props. If we push only the slug `router.push('route')`, then the props will be requested once. If there is a way out of this situation except how to intercept `_next/data/{route}.json` url with .html part in route and delete it before nextjs handled request?
